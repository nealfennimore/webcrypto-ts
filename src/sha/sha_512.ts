import * as alg from "../alg";
import { WebCrypto } from "../crypto";
import { ShaShared } from "./shared";

export namespace SHA_512 {
    export const digest = (data: BufferSource) =>
        WebCrypto.digest(alg.SHA.Variant.SHA_512, data);

    export const hexify = ShaShared.hexify;
}
