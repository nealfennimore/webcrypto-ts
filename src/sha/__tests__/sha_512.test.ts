import { SHA_512 } from "../sha_512";

describe("SHA_512", () => {
    it("should hash", async () => {
        const data = encode("a message");
        const hash = await SHA_512.digest(data);
        expect(hash.byteLength).toEqual(64);
    });
    it("should hexify", async () => {
        const data = encode("a message");
        const hash = await SHA_512.digest(data);
        expect(SHA_512.hexify(hash)).toMatchSnapshot();
    });
});
