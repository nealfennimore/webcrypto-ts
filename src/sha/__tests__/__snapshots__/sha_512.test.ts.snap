// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SHA_512 should hexify 1`] = `"35d655e1c7ba8419feb0510dffc953c36e87b5c35de6a7d03211897a0d4e35251562625fb71fba29ead1453bf2865d00ff668f0f980318367b2303f81c3a7a26"`;
